src/helpers/files.js

```diff
@@ -202,64 +200,45 @@ export const FilesWorkspaceHeader = new Header({
 	enabled(_, view) {
 		return ['files', 'favorites', 'public-share'].includes(view.id)
 	},
-
-	async render(el, folder, view) {
-		if (vm) {
-			// Enforce destroying of the old rendering and rerender as the FilesListHeader calls render on every folder change
-			vm.$destroy()
-			vm = null
+	render: async (el, folder) => {
+		// If the RichWorkspace component is already mounted, destroy it
+		if (!FilesHeaderRichWorkspaceView) {
+			FilesHeaderRichWorkspaceView = (
+				await import('../views/RichWorkspace.vue')
+			).default
+		} else {
+			FilesHeaderRichWorkspaceInstance.$destroy()
+			console.debug('Destroying existing FilesHeaderRichWorkspaceInstance')
 		}
-		const hasRichWorkspace =
-			!!folder.attributes['rich-workspace-file'] || !!newWorkspaceCreated
-		const path = newWorkspaceCreated
-			? dirname(newWorkspaceCreated.path)
-			: folder.path
-		const content = newWorkspaceCreated
-			? ''
-			: folder.attributes['rich-workspace']
-
-		newWorkspaceCreated = false
-
-		const { default: RichWorkspace } = await import(
-			'./../views/RichWorkspace.vue'
-		)
 
-		import('vue').then((module) => {
-			el.id = 'files-workspace-wrapper'
-
-			// Todo: remove this hack
-			const Vue = module.default
-			Vue.prototype.t = window.t
-			Vue.prototype.n = window.n
-			Vue.prototype.OCA = window.OCA
-
-			const View = Vue.extend(RichWorkspace)
-			vm = new View({
-				propsData: {
-					path,
-					hasRichWorkspace,
-					content,
-				},
-			}).$mount(el)
-		})
+		const hasRichWorkspace = !!folder.attributes['rich-workspace-file']
+		const content = folder.attributes['rich-workspace'] || ''
+		const path = folder.path || ''
+
+		// Create a new instance of the RichWorkspace component
+		FilesHeaderRichWorkspaceInstance = new Vue({
+			extends: FilesHeaderRichWorkspaceView,
+			propsData: {
+				content,
+				hasRichWorkspace,
+				path,
+			},
+		}).$mount(el)
+
+		window.FilesHeaderRichWorkspaceInstance = FilesHeaderRichWorkspaceInstance
 	},
 
-	updated(folder, view) {
-		newWorkspaceCreated = false
-
-		if (!vm) {
-			console.warn('No vue instance found for FilesWorkspaceHeader')
+	updated(folder) {
+		if (!FilesHeaderRichWorkspaceInstance) {
+			console.error('No vue instance found for FilesWorkspaceHeader')
 			return
 		}
 
-		// Currently there is not much use in updating the vue instance props since render is called on every folder change
-		// removing the rendered element from the DOM
-		// This is only relevant if switching to a folder that has no content as then the render function is not called
-
 		const hasRichWorkspace = !!folder.attributes['rich-workspace-file']
-		vm.path = folder.path
-		vm.hasRichWorkspace = hasRichWorkspace
-		vm.content = folder.attributes['rich-workspace']
+		FilesHeaderRichWorkspaceInstance.hasRichWorkspace = hasRichWorkspace ?? false
```

Comment:
minor nitpick:
`hasRichWorkspace` as defined above is a boolean, right? So `?? false` does  nothing.

```suggestion
		FilesHeaderRichWorkspaceInstance.hasRichWorkspace = hasRichWorkspace
```